### ISSUE TITLE
fontBuilder minimal example does not work on windows

### ISSUE THREAD

**Issue by massaad** [fonttools/fonttools#1660]
I am just starting with font generation, and I am trying to generate a font through code, so fontBuilder seemed like the ideal solution for me. Unfortunately, I am not sure how to start... using the minimal example included in the code is not working for me on windows. Even though the font files are created, If I try to view it, I get a "The requested file sample.otf is not a valid font file."

Are there any steps to get that code generate a minimal font that is viewable?

Thanks.

**Comment by miguelsousa** [comment]
I suspect that your font doesn't have one of the `name` table IDs required by Windows.
My colleague @cjchapman recently did some tests on that and these were his observations:

> Windows — `name`  IDs 1,2,3,4,6 required (platform=Windows)
> Mac — `name` IDs 1,2,3,4,6 required (platform=Mac, if *any* Mac IDs, otherwise uses Windows IDs)
> variable fonts — `name` IDs 16 & 17 required per MS
>
> Note that in my own experiments in Windows 10, it seems that only `name` IDs 1,2,3,4 are required (not 6).

**Comment by justvanrossum** [comment]
Yeah, it seems the minimal example in the code misses 3 (`uniqueFontIdentifier`) and 4 (`fullName`).

Untested:

```python
from fontTools.fontBuilder import FontBuilder
from fontTools.pens.ttGlyphPen import TTGlyphPen

def drawTestGlyph(pen):
    pen.moveTo((100, 100))
    pen.lineTo((100, 1000))
    pen.qCurveTo((200, 900), (400, 900), (500, 1000))
    pen.lineTo((500, 100))
    pen.closePath()

fb = FontBuilder(1024, isTTF=True)
fb.setupGlyphOrder([".notdef", ".null", "A", "a"])
fb.setupCharacterMap({65: "A", 97: "a"})

advanceWidths = {".notdef": 600, "A": 600, "a": 600, ".null": 600}

familyName = "HelloTestFont"
styleName = "TotallyNormal"
nameStrings = dict(familyName=dict(en="HelloTestFont", nl="HalloTestFont"),
                   styleName=dict(en="TotallyNormal", nl="TotaalNormaal"),
                   uniqueFontIdentifier="ABC; HelloTestFont TotallyNormal",
                   fullName="HelloTestFont TotallyNormal")
nameStrings['psName'] = familyName + "-" + styleName

pen = TTGlyphPen(None)
drawTestGlyph(pen)
glyph = pen.glyph()
glyphs = {".notdef": glyph, "A": glyph, "a": glyph, ".null": glyph}
fb.setupGlyf(glyphs)

metrics = {}
glyphTable = fb.font["glyf"]
for gn, advanceWidth in advanceWidths.items():
    metrics[gn] = (advanceWidth, glyphTable[gn].xMin)
fb.setupHorizontalMetrics(metrics)

fb.setupHorizontalHeader(ascent=824, descent=200)
fb.setupNameTable(nameStrings)
fb.setupOS2()
fb.setupPost()

fb.save("test.ttf")
```

It would be great if you could test this and report back whether it helped.

**Comment by massaad** [comment]
This did not work for me, but I think we are heading in the right direction. I used the same suggestions to update the otf minimal example, and that worked. I included those changes in a pull request.

**Comment by massaad** [comment]
fixed in #1661
